### DISCUSSION TITLE
in process server works with one test more then one test it say name already exists

### DISCUSSION THREAD

**Discussion by buttonsquared** [LogNet/grpc-spring-boot-starter#321] [Q&A]
When I am writing tests I am adding to my properties
grpc:
  enabled: false
  inProcessServerName: testingServer

Everything works great.  But when I build with more the one test I get the error grpc server can't start server name already exists.  Is there a way to make it either shut down the server after each test or just start it once and if it's running not start it again?

**Comment by jvmlet** [comment]
Are you using spring boot test support in your tests ?

**Comment by jvmlet** [comment]
The simplest approach is to use  `@ActiveProfile("test")` on your test class and then have `application-test.yml` as 

````yml
grpc:
  enabled: false
  inProcessServerName: ${random.value}
````
Then, in order to create the channel, you inject `GRpcServerProperties` bean and create channel as seen [here](https://github.com/LogNet/grpc-spring-boot-starter/blob/e34454e567b1d83674147b8633071e37ccf1e0a1/grpc-spring-boot-starter-demo/src/test/java/org/lognet/springboot/grpc/GrpcServerTestBase.java#L94)
